### DISCUSSION TITLE
RabbitMQ Cluster and Operator upgrade

### DISCUSSION THREAD

**Discussion by tferfce** [rabbitmq/cluster-operator#1724] [Q&A] [answered]
Hello,

I'm facing a dilemma regarding upgrade. Currently we have a RabbitMQ Cluster (3.9.x) managed by k8s rmq operator (v2.5.0).
I want to upgrade the operator (v2.9.0) and RMQ Cluster (3.13.2). On the operator side it is indicating that when i upgrade the v2.9.0 it will automatically upgrade the Cluster itself to 3.13.2 [notes](https://github.com/rabbitmq/cluster-operator/releases/tag/v2.9.0) but i want to avoid that because to upgrade from 3.9.x version of RabbitMQ cluster to 3.13.2 it needs a series of events needed which i have them documented. [upgarde notes](https://www.rabbitmq.com/docs/upgrade)

We have an option to pause the reconciliation of the operator and proceed with RMQ Cluster upgrade but then whatever change i make on Kind: RabbitmqCluster wont have any effect on the pods for step by step upgrades and i need to change the image on the statefulset that the operator produces seeing the RabbitmqCluster. The thing is that i don't know what other changes the operator will introduce on the statefulset other than the image update so changing the statefulset might not be enough and compatible with operator.

Also another option is to keep v2.5.0 operator version and upgrade the cluster step by step, and when i get to latest cluster version proceed with upgrading the operator. The problem is that i cant find a compatibility matrix for operators and clusters.

Any ideas or proposal?

Thanks,
Thomas

```
To go from 3.9.x -> 3.13.2

3.9.x -> 3.10.25 (image: 3.10.25-management)

3.10.x -> 3.11.x (image: 3.11.28-management)
    Feature Flags to be enabled (DONE)
        user_limits
        maintenance_mode_status
        implicit_default_bindings
        quorum_queue
        virtual_host_metadata

3.11.x -> 3.12.x (All feature flags must be enabled before the upgrade) (image: 3.12.14-management)
    Feature Flags to be enabled (PENDING)
        stream_queue
        classic_queue_type_delivery_support
        tracking_records_in_ets
        stream_single_active_consumer
        listener_records_in_ets
        feature_flags_v2
        direct_exchange_routing_v2
        classic_mirrored_queue_version

3.12.x -> 3.13.2 (image: 3.13.2-management)
```

**Comment by mkuratczyk** [comment, marked as answer]
Upgrading the Operator will not upgrade the cluster to 3.13. It will update the `StatefulSet`, since some of the STS-level settings changed. 

Also, do not upgrade to 3.13.2 - upgrade to 3.13.7.
